src/vs/workbench/contrib/workspace/browser/workspace.contribution.ts

```diff
@@ -344,43 +352,41 @@ export class WorkspaceTrustUXHandler extends Disposable implements IWorkbenchCon
 	//#region Dialog
 
 	private async doShowModal(question: string, trustedOption: { label: string; sublabel: string }, untrustedOption: { label: string; sublabel: string }, markdownStrings: string[], trustParentString?: string): Promise<void> {
-		const result = await this.dialogService.show(
-			Severity.Info,
-			question,
-			[
-				trustedOption.label,
-				untrustedOption.label,
-			],
-			{
-				checkbox: trustParentString ? {
-					label: trustParentString
-				} : undefined,
-				custom: {
-					buttonDetails: [
-						trustedOption.sublabel,
-						untrustedOption.sublabel
-					],
-					disableCloseAction: true,
-					icon: Codicon.shield,
-					markdownDetails: markdownStrings.map(md => { return { markdown: new MarkdownString(md) }; })
+		await this.dialogService.prompt({
```

Comment:
@lszomoru fyi please check, this one is a bit more complex adoption of the new dialog api